ports/geckolib/properties.mako.rs

```diff
@@ -317,9 +317,10 @@ def set_gecko_property(ffi_name, expr):
                                &mut self.gecko.${union_ffi_name});
     }
     fn copy_${ident}_from(&mut self, other: &Self) {
-        debug_assert_unit_is_safe_to_copy(self.gecko.${unit_ffi_name});
+        unsafe { self.gecko.${union_ffi_name}.reset(&mut self.gecko.${unit_ffi_name}) };
         self.gecko.${unit_ffi_name} =  other.gecko.${unit_ffi_name};
         self.gecko.${union_ffi_name} = other.gecko.${union_ffi_name};
+        unsafe { self.gecko.${union_ffi_name}.addref_opt(&self.gecko.${unit_ffi_name}) };
```

Comment:
Rather than assigning directly to the unit and union, and then calling `addref_opt` (or `addref_if_calc` if renamed), we could just have a single call to a `set` function on `nsStyleUnion` that takes a reference to the unit as an argument, and does the unit check and addref if necessary.  Currently it's kind of splitting the responsibility of dealing with different units correctly across `nsStyleUnion` functions and code here poking directly into the unit/union.